Bug: `no-useless-default-assignment` false positive when `exactOptionalPropertyTypes: true`

cd to a temp dir, run:
```bash
npm install --save-dev eslint typescript typescript-eslint oxlint oxlint-tsgolint
npx tsc --init
```
create `eslint.config.mjs`:
```js
import { defineConfig } from "eslint/config"
import tseslint from "typescript-eslint"
export default defineConfig({
  files: ["test.ts"],
  extends: [tseslint.configs.base],
  rules: {
    "@typescript-eslint/no-useless-default-assignment": "error",
  },
  languageOptions: {
    parserOptions: {
      projectService: true,
    },
  },
})
```
create `.oxlintrc.json`:
```jsonc
{
  "categories": {
    "correctness": "off"
  },
  "rules": {
    "typescript/no-useless-default-assignment": "error"
  }
}
```
create `test.ts`:
```ts
function test(options?: { offset?: number }) {
  const { offset = 5 } = { ...options }
  offset.toString()
}
```

### With `exactOptionalPropertyTypes: false`

Modify the generated `tsconfig.json` and set `exactOptionalPropertyTypes: false`
```bash
npx eslint test.ts
# No error 

npx tsc
# No error 

npx oxlint --type-aware test.ts
# No error 
```

### With `exactOptionalPropertyTypes: true`

Modify `tsconfig.json` and set `exactOptionalPropertyTypes: true`
```bash
npx eslint test.ts
# No error 

npx tsc
# No error 

npx oxlint --type-aware test.ts
# False positive:
#   typescript-eslint(no-useless-default-assignment): Default value is useless because the property has type `number` (not nullish). This default assignment will never be used.
```

Then, if applying the fix provided by oxlint, tsc will report error
```bash
npx oxlint --type-aware --fix test.ts
# No error

npx eslint test.ts
# No error

npx tsc
# `test.ts:3:3 - error TS18048: 'offset' is possibly 'undefined'.`

```

--- a/internal/rules/no_useless_default_assignment/no_useless_default_assignment.go
+++ b/internal/rules/no_useless_default_assignment/no_useless_default_assignment.go
@@ -171,20 +171,6 @@ func getAssignmentTargetNode(node *ast.Node) *ast.Node {
 	return nil
 }
 
-func hasConditionalInitializer(node *ast.Node) bool {
-	if node == nil || node.Parent == nil {
-		return false
-	}
-
-	parent := node.Parent
-	if ast.IsVariableDeclaration(parent) && parent.Initializer() != nil {
-		initializer := ast.SkipParentheses(parent.Initializer())
-		return ast.IsConditionalExpression(initializer) || ast.IsLogicalExpression(initializer)
-	}
-
-	return hasConditionalInitializer(parent)
-}
-
 func hasPropertyInAllBranches(expression *ast.Node, propertyName string) bool {
 	expression = ast.SkipParentheses(expression)
 	if ast.IsObjectLiteralExpression(expression) {
@@ -256,10 +242,12 @@ var NoUselessDefaultAssignmentRule = rule.Rule{
 				}
 
 				if utils.IsSymbolFlagSet(symbol, ast.SymbolFlagsOptional) {
-					if parent := parentPattern.Parent; ast.IsVariableDeclaration(parent) && parent.Initializer() != nil && hasConditionalInitializer(parentPattern) {
+					if parent := parentPattern.Parent; ast.IsVariableDeclaration(parent) && parent.Initializer() != nil {
 						if !hasPropertyInAllBranches(parent.Initializer(), propertyName) {
 							return nil
 						}
+					} else {
+						return nil
 					}
 				}
 

--- a/internal/rules/no_useless_default_assignment/no_useless_default_assignment_test.go
+++ b/internal/rules/no_useless_default_assignment/no_useless_default_assignment_test.go
@@ -27,6 +27,10 @@ func TestNoUselessDefaultAssignmentRule(t *testing.T) {
 			Code: "\n      const obj: { a?: string } = {};\n      const { a = 'default' } = obj;\n    ",
 		},
 		{
+			Code:     "\n      function test(options?: { offset?: number }) {\n        const { offset = 5 } = { ...options };\n        offset.toString();\n      }\n    ",
+			TSConfig: "tsconfig.exactOptionalPropertyTypes.json",
+		},
+		{
 			Code: "\n      function test(a: string | undefined = 'default') {\n        return a;\n      }\n    ",
 		},
 		{

